src/python/tests/test_maf_to_fasta.py

```diff
@@ -0,0 +1,180 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `maf_to_fasta.py` script.
+
+Typical usage example::
+
+    $ pytest test_maf_to_fasta.py
+
+"""
+
+from contextlib import nullcontext as does_not_raise
+import filecmp
+from importlib.abc import Loader
+from importlib.machinery import ModuleSpec
+from importlib.util import module_from_spec, spec_from_file_location
+from pathlib import Path
+import re
+import sys
+from types import ModuleType
+from typing import ContextManager, Iterable, List, Pattern, Union
+
+import pytest
+from pytest import raises
+
+
+def import_module_from_file(module_file: Union[Path, str]) -> ModuleType:
+    """Import module from file path.
+
+    The name of the imported module is the basename of the specified module
+    file without its extension.
+
+    In addition to being returned by this function, the imported module is
+    loaded into the sys.modules dictionary, allowing for commands such as
+    :code:`from <module> import <class>`.
+
+    Args:
+        module_file: File path of module to import.
+
+    Returns:
+        The imported module.
+
+    """
+    if not isinstance(module_file, Path):
+        module_file = Path(module_file)
+    module_name = module_file.stem
+
+    module_spec = spec_from_file_location(module_name, module_file)
+
+    if not isinstance(module_spec, ModuleSpec):
+        raise ImportError(f"ModuleSpec not created for module file '{module_file}'")
+    if not isinstance(module_spec.loader, Loader):
+        raise ImportError(f"no loader found for module file '{module_file}'")
+
+    module = module_from_spec(module_spec)
+    sys.modules[module_name] = module
+    module_spec.loader.exec_module(module)
+
+    return module
+
+
+script_path = Path(__file__).parents[3] / 'scripts' / 'hal_alignment' / 'maf_to_fasta.py'
+import_module_from_file(script_path)
+
+
+# pylint: disable=import-error,wrong-import-order,wrong-import-position
+
+import maf_to_fasta  # type: ignore
+
+# pylint: enable=import-error,wrong-import-order,wrong-import-position
+
+
+class TestMafToFasta:
+    """Tests script maf_to_fasta.py"""
```

Comment:
```suggestion
    """Tests script `maf_to_fasta.py`"""
```

Reply:
Thanks, I will do this, though I might use double backticks to be on the safe side.

Reply:
Sure! I was just following what has been done in other tests.

Reply:
Fair point, it's possible I'm being overly cautious here.